torch/_utils.py

```diff
@@ -203,6 +203,16 @@ def set_tensor_metadata(tensor, metadata):
     torch._C._set_tensor_metadata(tensor, metadata)  # type: ignore[attr-defined]
 
 
+def _restore_device_fake_mode(tensor):
+    if torch._guards.detect_fake_mode(None) is not None:
+        if tensor.untyped_storage()._fake_device is not None:
+            device = _get_restore_location(tensor.untyped_storage()._fake_device)
```

Comment:
Why do you delay this call until now instead of doing it when restore_location happens below?

Reply:
At this point we have access to the tensor that owns the storage where in the file below we only had access to the storage without knowing which tensor owned it

Also the restore_location below would try to move the storage to device, this helper just gets us the new str/torch.device